src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -1823,6 +1843,50 @@ void enableSSL(String host, int port) throws SQLServerException
 		}
 	}
 	
+	/**
+	 * Checking if FIPS is enabled or not. 
+	 * @param fipsProvider FIPS Provider
+	 * @return {@code true} if Customer wants to use FIPS Provider. 
+	 * @since 6.1.2
+	 */
+	private boolean isFipsEnabled(String fipsProvider, String trustStoreType) {
+		boolean isFipsEnabled = false;
+		boolean isEncryptOn;
+		boolean isValidTrustStoreType;
+		boolean isTrustServerCertificate;
+
+		if (!StringUtils.isEmpty(fipsProvider)) {
+			isEncryptOn = TDS.ENCRYPT_ON == con.getRequestedEncryptionLevel();
+
+			//Here different FIPS provider supports different KeyStore type along with different JVM Implementation. 
+			isValidTrustStoreType = !StringUtils.isEmpty(trustStoreType);
+			isTrustServerCertificate = con.trustServerCertificate();
+
+			if (isEncryptOn && isValidTrustStoreType && !isTrustServerCertificate) {
+				isFipsEnabled = true;
+			} else {
+				//If FIPS is not enabled issue warning and fall-back to non-FIPS mode.
+				StringBuilder sb = new StringBuilder();
+				sb.append("Could not switch to FIPS mode due to above settings: ");
```

Comment:
We cannot build strings directly in the implementation - it needs to be localizable to languages other than English.  Can we add a parameterized string resource and use that (you can see what is done for the error messages, for example)?